client/routes/workflow/components/cytoscape-graph.vue

```diff
@@ -0,0 +1,237 @@
+
+        //We are viewing a child workflow, show parent btn
+        if (previousExecutionRunId) {
+          store.commit("previousExecutionRoute", previousExecutionRunId);
+        } else if (parentWorkflowExecution) {
+          store.commit("parentRoute", parentWorkflowExecution);
+        }
+
+        this.nodes.push({
+          data: {
+            id: event.eventId,
+            name: event.eventType,
+            childRoute: childRoute,
+            newExecutionRunId: newExecutionRunId,
+            status: status
+          }
+        });
+      });
+      //Set the direct and inferred relationships
+      this.events.forEach(node => {
+        this.setDirectAndInferred(node);
+      });
+
+      //Set the chronological relationships.
+      //If the node is not referred to as a parent it should be connected back to the graph with a chron child
+      this.events.forEach(node => {
+        if (!this.parentArray.includes(node.eventId)) {
+          this.setChron(node);
+        }
+      });
+    },
+    setDirectAndInferred(node) {
+      let nodeId = node.eventId,
+        { parent, inferredChild } = getEventConnections(node, this.events);
+      if (parent) {
+        this.parentArray.push(parent);
+        this.edges.push({
+          data: { source: parent, target: nodeId, type: "direct" }
+        });
+      }
+      if (inferredChild) {
+        this.parentArray.push(nodeId);
+        this.edges.push({
+          data: { source: nodeId, target: inferredChild, type: "inferred" }
+        });
+      }
+    },
+    setChron(node) {
+      let nodeId = node.eventId,
+        { chronologicalChild } = getEventConnections(node, this.events);
+      if (chronologicalChild) {
+        this.edges.push({
+          data: {
+            source: nodeId,
+            target: chronologicalChild,
+            type: "chronological"
+          }
+        });
+      }
+    },
+    async viewInit() {
+      let container = this.$refs.cy;
+      let cy = (window.cy = cytoscape({
+        autoungrabify: true,
+        styleEnabled: true,
+        container: container,
+        headless: true,
+        hideEdgesOnViewport: true,
+        //Uncomment the two lines below for better performance
+        //textureOnViewport: true,
+        //pixelRatio: 1,
+        style: graphStyles,
+        elements: {
+          nodes: this.nodes,
+          edges: this.edges
+        },
+        layout: {
+          name: "dagre",
+          nodeDimensionsIncludeLabels: true,
+          spacingFactor: 1.2, // to avoid node collision
+          nodeSep: 230,
+          edgeSep: 100,
+          rankSep: 70
+        }
+      }));
+      cy.on("mouseover", "node", function(e) {
+        container.style.cursor = "pointer";
+      });
+      cy.on("mouseout", "node", function(e) {
+        container.style.cursor = "default";
+      });
+
+      let self = this;
+
+      //Register click event
+      cy.on("tap", function(evt) {
```

Comment:
have you tried arrow functions here? it would avoid having to reference `self = this`.
See [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions)